// @JDB/build/core-publisher/config.js
var path = require('path');
var packageConfig = null;
var packageConfigPath = path.join(process.cwd(), './package.json');
// 读取用户项目的配置文件
try {
    packageConfig = require(packageConfigPath);
} catch (error) {
    // 文件不存在则直接中断构建
    console.log('[FATAL ERROR]', 'Please add a package.json file to your project!');
    console.log('[FATAL ERROR]', 'package.json path: ', packageConfigPath);
    process.exit(1);
}

var packageName = packageConfig.name || 'site';
var dest = './dist/core';
var libSrc = [
    'lib/promise/promise.js',
    'lib/blueimp-md5/js/md5.min.js',
    'lib/ionic/js/ionic.bundle.min.js',
    'lib/js-url/url.min.js',
    'lib/microtemplates/index.js',
    'lib/ngCordova/ng-cordova.js',
    'lib/zepto.js/index.js',
    'lib/angular-carousel/angular-carousel.js',
    'lib/angular-cookie/angular-cookie.js',
    'lib/angular-jdb/angular-jdb.js',
    'lib/angular-route/angular-route.js',
    'lib/angular-sanitize/angular-sanitize.js',
    'lib/angular-touch/angular-touch.js',
    'lib/angular-ui-router/angular-ui-router.js',
    'lib/createjs/easeljs-0.8.2.min.js'
];
var commonSrc = [
    'common/js/util.js',
    'common/dialog/dialog.js',
    'common/ng-infinite-scroll.js'
];
var iosCordovaSrc = ['cordova_ios/cordova.js'];
var androidCordovaSrc = [
    'cordova_android/cordova.js',
    'cordova_android/cordova_plugins.js',
    'cordova_android/plugins/**/*.js'
];

var jdbjsSrc = ['jdb/components/*/**/*.js'];

var config = {
    dest: dest,
    localDest: {
        general: './dist/general',
        android: './dist/android',
        ios: './dist/iOS'
    },
    styles: {
        src: [
            'common/css/base.css',
            'common/dialog/dialog.scss',
            'lib/ionic/css/ionic.css',
            'jdb/components/*/**/*.{scss,css}'
        ],
        dest: dest
    },
    images: {
        src: ['jdb/components/*/images/**/*.{jpg,png,gif}'],
        dest: dest + '/images'
    },
    scripts: {
        srcWithoutCordova: [].concat(libSrc, commonSrc, jdbjsSrc),
        iosSrc: [].concat(libSrc, iosCordovaSrc, commonSrc, jdbjsSrc),
        androidSrc: [].concat(libSrc, androidCordovaSrc, commonSrc, jdbjsSrc),
        dest: dest
    }
};

module.exports = config;
